### PR TITLE
Committing the changes for data model binding.

### PR DESCRIPTION
**Pull request by ashish-compro**

### CODE REVIEW

File: client/app/shared/text-box/text-box.component.ts [hunk 1]
```diff
@@ -10,12 +10,13 @@ import { itemSchema, itemDataModel } from '../UIConfig.model';
 export class TextBoxComponent extends BaseComponent implements OnInit {
   labelConfig: itemSchema = new itemSchema();
   itemDataModel;
-  modelRef;
+  modelRef = { value: "" };
 
   ngOnInit() {
     this.itemDataModel = itemDataModel;
```
Comment (ashish-compro) [inline client/app/shared/text-box/text-box.component.ts]: Need to remove the if check as there will be no case when we won't have the model reference for text box component.

File: client/app/step-builder/shared/builder-model.service.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    setModel(newModel) {
+        this.builderModel = newModel;
+    }
+
+    getModel() {
+        return this.builderModel;
+    }
+
+    getModelRef(modelRefStr) {
+        return eval("this.builderModel." + modelRefStr.replace(/['{','}']/g, ""));
```
Comment (ashish-compro) [inline client/app/step-builder/shared/builder-model.service.ts]: Change the tokens from '{' to '{{'.
Rename the function name to getCompModel

File: server/controllers/db.controller.js [hunk 3]
```diff
@@ -45,6 +45,6 @@ function getFilePath(templateId, fileKey, callback) {
     filterCriteria[fileKey] = true;
 
     skillConfigRepoModel.getFilePath({"template_id": templateId}, filterCriteria, (error, data) => {
-        callback(data[0][fileKey], error);
+        callback("filestore/skills/"+data[0][fileKey], error);
```
Comment (vsingb3) [inline server/controllers/db.controller.js]: Not sure about this.

**Review by vsingb3 — APPROVED** [review]
Approving to enable resolve conflict option

**Pull request merged**